src/lua/LuaBody.cpp

```diff
@@ -636,22 +654,20 @@ static int l_body_get_ground_position(lua_State *l)
 	}
 
 	Frame *f = Frame::GetFrame(b->GetFrame());
-	if (!f->IsRotFrame())
-		return 0;
 
+	if (!f)
+	{
+		lua_pushnil(l);
+		return 1;
+	}
 	vector3d pos = b->GetPosition();
 	double latitude = atan2(pos.y, sqrt(pos.x * pos.x + pos.z * pos.z));
 	double longitude = atan2(pos.x, pos.z);
 	lua_pushnumber(l, latitude);
 	lua_pushnumber(l, longitude);
 	Body *astro = f->GetBody();
-	if (astro->IsType(ObjectType::TERRAINBODY)) {
-		double radius = static_cast<TerrainBody *>(astro)->GetTerrainHeight(pos.Normalized());
-		double altitude = pos.Length() - radius;
-		lua_pushnumber(l, altitude);
-	} else {
-		lua_pushnil(l);
-	}
+	double altitude = b->GetAltitudeRelTo(astro);
+	lua_pushnumber(l, altitude);
```

Comment:
This function should take a parameter to pass to `GetAltitudeRelTo` similar to `body:GetAltitudeRelTo()`.